content/english/blog/globally_distributed_webassembly_applications_with_wasmcloud_and_nats.md

```diff
@@ -0,0 +1,364 @@
+
+  
+
+[^1]: NATS also accepts [connections](https://docs.nats.io/nats-concepts/connectivity) over TLS, WebSockets, and MQTT
+
+  
+
+# wasmCloud and NATS
+
+wasmCloud uses NATS in a multitude of ways. To name a few:
+
+1.  Request/reply messaging for remote procedure calls between actors and capability providers. Additionally this is used for the [control interface](https://github.com/wasmCloud/control-interface-client/), which allows for remote managing of actors, providers, and hosts via [wash](https://github.com/wasmCloud/wash).
+2.  Jetstream to persist link definitions and claims so that they are durable and automatically delivered to new wasmCloud hosts joining an existing lattice
+3.  Communication with a wasmCloud configuration service and the soon-to-come application deployment manager ([wadm](https://github.com/wasmCloud/wadm))
+
+  
+
+# Gathering Prerequisites
+
+This example requires a few prerequisites:
+
+*   The components in the wasmCloud [installation guide](https://wasmcloud.dev/overview/installation/), which includes `wash` , the wasmCloud host runtime, and the `nats-server` binary.
+*   NATS account credentials to access NGS (we'll walk through this below)
+
+We're also going to use a few files from the aptly named [ngs](https://github.com/wasmCloud/examples/tree/main/ngs) folder in our examples repository to deploy a Wasm microservice (we call this an [actor](https://wasmcloud.dev/reference/host-runtime/actors/)) that securely fetches a random image of a cat or a dog. Later on we're going to use some files in this folder, you can either clone this repository or just copy and paste as we go along.
+
+To help illustrate the architecture of our application, take a look at this diagram:
+
+![](../../images/blogs/ngs-global/excalidraw.png)
+
+We'll have wasmCloud running both locally and in the cloud (or just on another machine), and we'll be spreading compute across these two wasmCloud hosts. Don't worry, you won't have to look up your local IP address or expose any ports, NATS makes distributed computing a breeze.
+
+  
+
+The first step will be to get yourself a set of NGS credentials. Navigate to [https://app.ngs.global](https://app.ngs.global) and select "Try It Out" under **Free**.
+![](../../images/blogs/ngs-global/ngs-signup.png)
+
+For simplicity, go ahead and name this account `wasmcloud` to keep it separate from any other NATS accounts you may create in the future.
```

Comment:
maybe we don't want people to call their accounts wasmcloud? Should we suggest they name it something meaningful to them and just point out that for the duration of this blog post, we'll be calling _ours_ `wasmcloud` ?

Reply:
The main reason why I pointed that out was to create an easy separation and to keep the file paths consistent later on in the post when I reference where their account creds are

Reply:
can they rename the account later? If so, we should mention that so that we don't end up having all of our users creating a mountain of `wasmcloud` accounts confusing both them and NGS